src/core/__tests__/ReactErrorBoundaries-test.js

```diff
@@ -236,4 +236,353 @@ describe('ReactErrorBoundaries', function() {
+        log.push('Inquisitive componentDidMount');
+      }
+      componentWillUnmount() {
+        log.push('Inquisitive componentWillUnmount');
+      }
+    }
+
+    class Angry extends React.Component {
+      render() {
+        log.push('Angry render');
+        throw new Error('Please, do not render me.');
+      }
+      componentDidMount() {
+        log.push('Angry componentDidMount');
+      }
+      componentWillUnmount() {
+        log.push('Angry componentWillUnmount');
+      }
+    }
+
+    var container = document.createElement('div');
+    ReactDOM.render(<Box angry={false} />, container);
+    ReactDOM.render(<Box angry={true} />, container);
+    expect(container.textContent).toBe('Error: Please, do not render me.');
+    expect(log).toEqual([
+      'Box render',
+      'Inquisitive render',
+      'Inquisitive componentDidMount',
+      'Inquisitive ref [object Object]',
+      'Box componentDidMount',
+      'Box render',
+      'Inquisitive ref null',
+      'Inquisitive render',
+      'Angry render',
+      'error handled',
+      'Inquisitive ref null',
+      'Inquisitive componentWillUnmount',
+      'Box renderError',
+    ]);
+  });
+
+  it('catches componentWillUnmount errors on update', function() {
+    var log = [];
+
+    class ErrorBoundary extends React.Component {
+      constructor(props) {
+        super(props);
+        this.state = {errorMessage: null};
+      }
+      render() {
+        if (this.state.errorMessage != null) {
+          log.push('Box renderError');
+          return <div>Error: I am now a sad component :(</div>;
+        }
+        log.push('Box render');
+
+        return (
+          <div>
+            <BrokenUnmount />
+            <BrokenUnmount />
+            {this.props.angry ? null : <BrokenUnmount />}
+          </div>
+        );
+      }
+      unstable_handleError(e) {
+        log.push('error handled');
+        this.setState({errorMessage: e.message});
+      }
+      componentDidMount() {
+        log.push('Box componentDidMount');
+      }
+      componentWillUnmount() {
+        log.push('Box componentWillUnmount');
+      }
+    }
+
+    class BrokenUnmount extends React.Component {
+      render() {
+        return <div>text</div>;
+      }
+      componentWillUnmount() {
+        log.push('BrokenUnmount is attempting to unmount');
+        throw new Error('Always broken.');
+      }
+    }
+
+    var container = document.createElement('div');
+    ReactDOM.render(<ErrorBoundary angry={false} />, container);
+    ReactDOM.render(<ErrorBoundary angry={true} />, container);
+    expect(container.textContent).toBe('Error: I am now a sad component :(');
+    expect(log).toEqual([
+      'Box render',
+      'Box componentDidMount',
+      'Box render',
+      'BrokenUnmount is attempting to unmount',
+      'error handled',
+      'BrokenUnmount is attempting to unmount',
+      'BrokenUnmount is attempting to unmount',
+      'BrokenUnmount is attempting to unmount',
```

Comment:
A little unfortunate that we re-call componentWillUnmount on the component that already failed to unmount.